From 0baaa7df48d9c7d055bc45074147cb305210246b Mon Sep 17 00:00:00 2001
From: Raviteja Narayanam <raviteja.narayanam@xilinx.com>
Date: Tue, 8 Sep 2020 17:15:17 +0530
Subject: [PATCH 1578/1852] i2c: xiic: Use 'nmsgs' variable instead of
 repeated_start

commit d878ccecf8a3a3ba561d2f410518b366efc10415 from
https://github.com/Xilinx/linux-xlnx.git

Remove the redundant repeated_start variable as it is setup at
different places, and read at only one place which can be decided
by 'nmsgs' variable alone.

Signed-off-by: Raviteja Narayanam <raviteja.narayanam@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/i2c/busses/i2c-xiic.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/drivers/i2c/busses/i2c-xiic.c b/drivers/i2c/busses/i2c-xiic.c
index 4deed040e05a..ad4460747823 100644
--- a/drivers/i2c/busses/i2c-xiic.c
+++ b/drivers/i2c/busses/i2c-xiic.c
@@ -60,7 +60,6 @@ enum xiic_endian {
  * @endianness: big/little-endian byte order
  * @clk:	Pointer to struct clk
  * @dynamic: Mode of controller
- * @repeated_start: Repeated start operation
  * @prev_msg_tx: Previous message is Tx
  */
 struct xiic_i2c {
@@ -78,7 +77,6 @@ struct xiic_i2c {
 	enum xiic_endian	endianness;
 	struct clk *clk;
 	bool dynamic;
-	bool repeated_start;
 	bool prev_msg_tx;
 };
 
@@ -441,7 +439,7 @@ static void xiic_std_fill_tx_fifo(struct xiic_i2c *i2c)
 
 	if (len > fifo_space)
 		len = fifo_space;
-	else if (len && !(i2c->repeated_start))
+	else if (len && !(i2c->nmsgs > 1))
 		len--;
 
 	while (len--) {
@@ -755,7 +753,6 @@ static void xiic_start_recv(struct xiic_i2c *i2c)
 
 		/* Check if RSTA should be set */
 		if (cr & XIIC_CR_MSMS_MASK) {
-			i2c->repeated_start = true;
 			/* Already a master, RSTA should be set */
 			xiic_setreg8(i2c, XIIC_CR_REG_OFFSET, (cr |
 				     XIIC_CR_REPEATED_START_MASK) &
@@ -774,7 +771,6 @@ static void xiic_start_recv(struct xiic_i2c *i2c)
 
 		/* Write to Control Register,to start transaction in Rx mode */
 		if ((cr & XIIC_CR_MSMS_MASK) == 0) {
-			i2c->repeated_start = false;
 			xiic_setreg8(i2c, XIIC_CR_REG_OFFSET, (cr |
 				     XIIC_CR_MSMS_MASK)
 				     & ~(XIIC_CR_DIR_IS_TX_MASK));
@@ -849,7 +845,6 @@ static void xiic_start_send(struct xiic_i2c *i2c)
 		/* Check if RSTA should be set */
 		cr = xiic_getreg8(i2c, XIIC_CR_REG_OFFSET);
 		if (cr & XIIC_CR_MSMS_MASK) {
-			i2c->repeated_start = true;
 			/* Already a master, RSTA should be set */
 			xiic_setreg8(i2c, XIIC_CR_REG_OFFSET, (cr |
 				     XIIC_CR_REPEATED_START_MASK |
@@ -864,7 +859,6 @@ static void xiic_start_send(struct xiic_i2c *i2c)
 		xiic_std_fill_tx_fifo(i2c);
 
 		if ((cr & XIIC_CR_MSMS_MASK) == 0) {
-			i2c->repeated_start = false;
 
 			/* Start Tx by writing to CR */
 			cr = xiic_getreg8(i2c, XIIC_CR_REG_OFFSET);
-- 
2.31.1

